src/BenchmarkDotNet/Extensions/ProcessExtensions.cs

```diff
@@ -119,6 +120,12 @@ internal static void SetEnvironmentVariables(this ProcessStartInfo start, Benchm
             if (benchmarkCase.Job.Environment.Runtime is MonoRuntime monoRuntime && !string.IsNullOrEmpty(monoRuntime.MonoBclPath))
                 start.EnvironmentVariables["MONO_PATH"] = monoRuntime.MonoBclPath;
 
+            // corerun does not understand runtimeconfig.json files;
+            // we have to set "COMPlus_GC*" environment variables as documented in
+            // https://docs.microsoft.com/en-us/dotnet/core/run-time-config/garbage-collector
+            if (benchmarkCase.Job.Infrastructure.Toolchain is CoreRunToolchain _)
```

Comment:
I am afraid that setting all the env vars always might have a different effect compared to what we had so far (not doing this for default settings).

It should be enough to just check `Gc.HasChanges` for `true` to filter out `default` settings

```suggestion
            if (benchmarkCase.Job.Infrastructure.Toolchain is CoreRunToolchain _ && benchmarkCase.Job.Environment.Gc.HasChanges)
```